### DISCUSSION TITLE
Why is the IV written to the SHA1 Hasher of the MDC?

### DISCUSSION THREAD

**Discussion by phifty** [ProtonMail/go-crypto#132] [Q&A] [answered]
I wonder where this comes from? I can't find it in the RFC.

https://github.com/ProtonMail/go-crypto/blob/c6815a8c17ad6a69d71ff580d2dd6869049d643d/openpgp/packet/symmetrically_encrypted.go#L286

Kind regards

**Comment by twiss** [comment, marked as answer]
Hello :wave: The MDC simply hashes all the plaintext contents of the Symmetrically Encrypted Integrity Protected Data packet, including the "explicit IV". This is described here: https://datatracker.ietf.org/doc/html/draft-ietf-openpgp-crypto-refresh#section-5.13.1.

**Reply by phifty** [reply]
Thanks, I've missed that.
